pyiceberg/table/snapshots.py

```diff
@@ -247,12 +248,19 @@ def __str__(self) -> str:
         result_str = f"{operation}id={self.snapshot_id}{parent_id}{schema_id}"
         return result_str
 
-    def manifests(self, io: FileIO) -> List[ManifestFile]:
-        if self.manifest_list is not None:
-            file = io.new_input(self.manifest_list)
+    @staticmethod
+    @lru_cache
+    def _manifests(io: FileIO, manifest_list: Optional[str]) -> List[ManifestFile]:
```

Comment:
I don't think this is the best place to add the caching.

With the newly introduced delete operation we can produce multiple snapshots. Each snapshot will produce an unique manifest list, but within the list there is a fair chance that we point to the same manifests.

It would be better to move the caching to the module level, so we can cache across manifest-lists.

Reply:
> It would be better to move the caching to the module level, so we can cache across manifest-lists.

Do you mean move the cache to *manifest* level? If so, we can add the cache to manifest.fetch_manifest_entry(). That would be across manifest-lists too.

Reply:
I think it should not be part of the `Snapshot` class. This class gets recreated, but then it will also invalidate the cache. The idea behind the LRU cache is that you keep it around until it grows to a certain size, and then you start dropping the last used entries.

If we move it to the top of the file, the cache continues to work across snapshots:
```suggestion
def _manifests(io: FileIO, manifest_list: Optional[str]) -> List[ManifestFile]:
```

WDYT?